libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java

```diff
@@ -466,6 +467,21 @@ public static void initInternal(Context context) {
         if (!TextUtils.isEmpty(idpAppUrlScheme)) {
             INSTANCE.idpAppURIScheme = idpAppUrlScheme;
         }
+
+        // Kicks off a thread to fetch auth config for the current server.
+        (new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                final String loginServer = SalesforceSDKManager.getInstance().getLoginServerManager().getSelectedLoginServer().url;
```

Comment:
This runs at the end of `SalesforceSDKManager.init()` to update settings for the current server when the app starts up.

Reply:
Shouldn't we wait for that call to complete?
Could we use the same code as AuthConfigTask?

Reply:
There's no good way to accomplish that in this case since there's no action after `init()` completes. Using an `AsyncTask` in this scenario would be the same as a `Runnable`. We recommend that the `init()` call be made from the `Application` class of the consuming application to initialize the SDK components, after which the main screen starts up. This should give ample time for the `Runnable` to finish execution. The only deterministic way I can think of is to maybe do this on initialization of `LoginActivity` instead, but that might be too frequent.